sdv/constraints/utils.py

```diff
@@ -126,22 +129,26 @@ def check_nans_row(row):
         return 'None'
 
 
-def add_nans_column(table_data, list_column_names):
-    """Add a categorical column to the table_data indicating where NaNs are.
+def compute_nans_column(table_data, list_column_names):
+    """Compute a categorical column to the table_data indicating where NaNs are.
 
     Args:
         table_data (pandas.DataFrame):
             The table data.
         list_column_names (list):
             The list of column names to check for NaNs.
+
+    Returns:
+        A dict with the column name as key and the column indicating where NaNs are as value.
+        Empty dict if there are no NaNs.
     """
     nan_column_name = '#'.join(list_column_names) + '.nan_component'
-    nan_column = table_data[list_column_names].apply(check_nans_row, axis=1)
-    if not (nan_column == 'None').all():
-        table_data[nan_column_name] = nan_column
-        return True
-    else:
-        return False
+    nan_column_dict = {}
+    column = table_data[list_column_names].apply(get_nan_component_value, axis=1)
+    if not (column == 'None').all():
+        nan_column_dict[nan_column_name] = column
```

Comment:
why not just return a series? So if all the values are 'None' return `None`, else returns a series with the proper name and values

Reply:
Yes, actually I did this first but without returning None when there is no nans, that's why. It's better now, done in [8d39f6a](https://github.com/sdv-dev/SDV/pull/1399/commits/8d39f6a89cf3ba2480e3640d8b87131cd658e7a9)